docs/jwst/srctype/description.rst

```diff
@@ -1,47 +1,60 @@
-
 Description
 ============
-The Source Type (``srctype``) step in the calibration pipeline checks or sets
-whether a spectroscopic source should be treated as a point or extended object.
+The Source Type (``srctype``) step in the calibration pipeline attempts to
+determine whether a spectroscopic source should be considered to be a point
+or extended object, populating the "SRCTYPE" keyword with a value of either
+"POINT" or "EXTENDED."
 This information is then used in some subsequent spectroscopic processing
 steps.
 
 Depending on the JWST observing mode, the observer may have the option of
-designating a source type in the APT template for the observations. They have
+designating a source type in the APT template for the observation. They have
 the choice of declaring whether or not the source should be considered
 extended. If they don't know the character of the source, they can also
-choose a value of unknown. The observer's choice in the APT is passed along
-to DMS processing, which sets the value of the ``SRCTYPE`` keyword in the
-primary header of the level-1b (_uncal.fits) product that's used as input
-to the calibration pipeline. The ``SRCTYPE`` keyword may have values of
-``POINT``, ``EXTENDED``, or ``UNKNOWN``.
+choose a value of "UNKNOWN." The observer's choice in the APT is passed along
+to DMS processing, which sets the value of the "SRCTYPE" keyword in the
+primary header of the uncalibrated (_uncal.fits) product that's used as input
+to the calibration pipeline. The resulting "SRCTYPE" values can be
+"POINT", "EXTENDED", or "UNKNOWN."
```

Comment:
I'm sure that we have at least some input files that are just so old that they don't have the SRCTYPE keyword present. We shouldn't be getting any imaging mode exposures run through this step, so hopefully that's not happening. The code does cover the use case where SRCTYPE is not set at all (or doesn't exist). But you're right that the docs above do make it sound like SRCTYPE will always be present in the uncal files, which isn't necessarily the case (for example, I don't know what happens when the user makes no selection at all in APT - is the keyword left out of the header?). I'll update the docs to be more explicit.

Reply:
I think there might be modes which `srctype` does not cover, as in the example above, which will cause extract_1d to fail. Since the `srctype` step is pretty simple it's possible to write a unit test which creates an `ImageModel` for every spectroscopic modes, runs it through `srctype` and checks for the expected result.

Reply:
I was thinking that the ``extract_1d`` step probably needs to handle, in some way, the case where SRCTYPE is not set. I'm not sure what the appropriate default should be. Perhaps EXTENDED? The ``pathloss`` and ``barshadow`` steps, which also depend on knowing whether the source is point or extended, have that kind of logic in them (I think they default to EXTENDED).

Reply:
Would different steps need different default values for SRCTYPE?  If not, then it seems to me that the `srctype` step itself should set the default.

Reply:
`extract_2d`  also needs to know what the source type is. If it's not set it currently skips the wavelength zero point correction.

Reply:
OK, you've convinced me that the ``source_type`` step should be the one to always set some default value, rather than having that logic spread around in multiple other steps. I'll make it so. I think the safest default for EXP_TYPEs not already covered by the rules set by the CalWG is to use EXTENDED.